src/k8s/pkg/k8sd/types/cluster_config_validate_test.go

```diff
@@ -54,3 +55,47 @@ func TestValidateCIDR(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateLoadbalancer(t *testing.T) {
+	for _, tc := range []struct {
+		cidr      []string
+		expectErr bool
+	}{
+		{cidr: []string{"10.3.0.0/16"}},
+		{cidr: []string{"2001:0db8::/32"}},
+		{cidr: []string{"10.3.0.0/16", "2001:0db8::/32"}},
+		{cidr: []string{"10.3.0.10-10.3.0.20"}},
+		{cidr: []string{"10.3.0.32/28", "10.3.0.10-10.3.0.20"}},
+		{cidr: []string{"2001:0db8::0-2001:0db8::10"}},
+		{cidr: []string{"10.3.0.32/28", "2001:0db8::0-2001:0db8::10"}},
+		{cidr: []string{""}, expectErr: true},
+		{cidr: []string{"bananas"}, expectErr: true},
+		{cidr: []string{"fd01::/64,fd02::/64,fd03::/64"}, expectErr: true},
+		{cidr: []string{"10.3.0.10-10.3.0.300"}, expectErr: true},
+		{cidr: []string{"2001:0db8::0-2001:0db8::gg"}, expectErr: true},
+		{cidr: []string{"", "10.3.0.10-10.3.0.300"}, expectErr: true},
+	} {
+		t.Run(strings.Join(tc.cidr, ","), func(t *testing.T) {
+			t.Run("LoadBalancer", func(t *testing.T) {
+				g := NewWithT(t)
+				config := types.ClusterConfig{
+					Network: types.Network{
+						Enabled:     vals.Pointer(true),
+						PodCIDR:     vals.Pointer("10.2.0.0/16"),
+						ServiceCIDR: vals.Pointer("10.1.0.0/16"),
+					},
+					LoadBalancer: types.LoadBalancer{
+						Enabled: vals.Pointer(true),
+						CIDRs:   vals.Pointer(tc.cidr),
+					},
+				}
+				err := config.Validate()
+				if tc.expectErr {
+					g.Expect(err).To(HaveOccurred())
+				} else {
+					g.Expect(err).To(BeNil())
```

Comment:
Since you're using HaveOccured in one branch, let's use HaveOccured in the else branch.

```suggestion
				if tc.expectErr {
					g.Expect(err).Should(HaveOccurred())
				} else {
					g.Expect(err).ShouldNot(HaveOccurred())
```

Reply:
Fixed